deploy/infrastructure/utils/variables.py

```diff
@@ -249,6 +248,73 @@ def write_files(definitions: Dict[str, str]):
             write_file(tfvars_md_filename, content)
 
 
+def read_file(file_path: str) -> str:
+    """
+    Reads a file and returns its content
+
+    Arguments:
+        file_path: the relative path of the file to be read from
+    Returns:
+        the content of the file as a str
+    """
+    with open(file_path, "r") as file:
+        content = file.read()
+    return content
+
+
+def diff_files(definitions: Dict[str, str]) -> bool:
+    """
+    Generates the `variables.gen.tf` file content in memory and diffs it
+    against the `variables.get.tf` files found on disk
+
+    Arguments:
+        definitions: a dict where the keys are the terraform variable names,
+        and the valuies are the content of that file.
+    Returns:
+        true iff the generated content and locally stored content are equal
+        string values
+    """
+    for path, variables in PROJECT_VARIABLES.items():
+        # Generate variables.gen.tf definition
+        var_filename = join(path, GENERATED_VARIABLES_FILENAME)
+        generated_content = get_variables_gen_tf_content(variables, definitions)
+
+        try:
+            actual_content = read_file(var_filename)
+        except Exception as e:
+            print(f"Error reading {var_filename}: {e}")
+            return False
+
+        if generated_content != actual_content:
+            return False
+
+    return True
+
+
 if __name__ == "__main__":
+    parser = argparse.ArgumentParser(usage="python variables")
+    parser.add_argument(
+        "--diff",
```

Comment:
@barroco Is it fine to assume that if no argument is passed we default to the original behavior of the script, or should we be adding an additional flag for that?